File: app/src/main/java/com/imceits/aungtuntun/alephcodeassignment/data/EnumConverter.java
package com.imceits.aungtuntun.alephcodeassignment.data;

import android.text.TextUtils;

import androidx.room.TypeConverter;

class EnumConverter {

    @TypeConverter
 public static String fromStatusToString(Status status) {
     return status == null ? null : status.getText();
 }

 @TypeConverter
 public static Status getStatus(String status) {
        if (!TextUtils.isEmpty(status)) {
            return Status.BORROWED.getText().equals(status) ? Status.BORROWED : Status.RETURNED;
        }else
            return Status.RETURNED;
 }
}

File: app/src/main/java/com/imceits/aungtuntun/toolsbox/data/EnumConverter.kt
package com.imceits.aungtuntun.toolsbox.data

import androidx.room.TypeConverter

class EnumConverter {

    @TypeConverter
    public fun fromStatusToString(status: Status?) : String? {
        return status?.value
    }

    @TypeConverter
    public fun getStatus(status: String?) : Status {
        if (!status.isNullOrEmpty()) {
            return if (Status.BORROWED.value == status) Status.BORROWED else Status.RETURNED
        }else
            return Status.RETURNED
    }
}